agent/agent-profiler/agent-diagnostics-jfr/src/main/java/com/microsoft/applicationinsights/diagnostics/jfr/CGroupData.java

```diff
@@ -20,63 +23,110 @@
 
   // CPU usage limit
-  private final long cpuLimit; // /sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us
+  private long cpuLimit; // /sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us
 
   // CPU usage period
-  private final long cpuPeriod; // /sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us
-
-  @JsonCreator
-  public CGroupData(
-      @JsonProperty(value = "kmemLimit") long kmemLimit,
-      @JsonProperty(value = "memoryLimit") long memoryLimit,
-      @JsonProperty(value = "memorySoftLimit") long memorySoftLimit,
-      @JsonProperty(value = "cpuLimit") long cpuLimit,
-      @JsonProperty(value = "cpuPeriod", required = false) Long cpuPeriod) {
-    this.kmemLimit = kmemLimit;
-    this.memoryLimit = memoryLimit;
-    this.memorySoftLimit = memorySoftLimit;
-    this.cpuLimit = cpuLimit;
-
-    if (cpuPeriod == null) {
-      // No CGroup data was found for this value
-      this.cpuPeriod = CGROUP_DATA_ABSENT;
-    } else {
-      this.cpuPeriod = cpuPeriod;
-    }
-  }
+  private long cpuPeriod = CGROUP_DATA_ABSENT; // /sys/fs/cgroup/cpu,cpuacct/cpu.cfs_quota_us
 
   public long getKmemLimit() {
     return kmemLimit;
   }
 
+  public CGroupData setKmemLimit(long kmemLimit) {
+    this.kmemLimit = kmemLimit;
+    return this;
+  }
+
   public long getMemoryLimit() {
     return memoryLimit;
   }
 
+  public CGroupData setMemoryLimit(long memoryLimit) {
+    this.memoryLimit = memoryLimit;
+    return this;
+  }
+
   public long getMemorySoftLimit() {
     return memorySoftLimit;
   }
 
+  public CGroupData setMemorySoftLimit(long memorySoftLimit) {
+    this.memorySoftLimit = memorySoftLimit;
+    return this;
+  }
+
   public long getCpuLimit() {
     return cpuLimit;
   }
 
+  public CGroupData setCpuLimit(long cpuLimit) {
+    this.cpuLimit = cpuLimit;
+    return this;
+  }
+
   public long getCpuPeriod() {
     return cpuPeriod;
   }
+
+  public CGroupData setCpuPeriod(long cpuPeriod) {
+    this.cpuPeriod = cpuPeriod;
+    return this;
+  }
+
+  @Override
+  public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
+    return jsonWriter
+        .writeStartObject()
+        .writeLongField("kmemLimit", kmemLimit)
+        .writeLongField("memoryLimit", memoryLimit)
+        .writeLongField("memorySoftLimit", memorySoftLimit)
+        .writeLongField("cpuLimit", cpuLimit)
+        .writeLongField("cpuPeriod", cpuPeriod)
+        .writeEndObject();
+  }
+
+  public static CGroupData fromJson(JsonReader jsonReader) throws IOException {
+    return jsonReader.readObject(
+        reader -> {
+          CGroupData deserializedValue = new CGroupData();
+
+          while (reader.nextToken() != JsonToken.END_OBJECT) {
+            String fieldName = reader.getFieldName();
+            reader.nextToken();
+            // In this case field names are case-sensitive but this could be replaced with
+            // 'equalsIgnoreCase' to
+            // make them case-insensitive.
```

Comment:
I'm guessing these comparisons are always case-sensitive? (in which case not sure this comment is needed)
```suggestion
```

Reply:
same here

Reply:
sorry, I didn't follow, this looks like new code?